### ISSUE TITLE
Instructions on how to use with a flake

### ISSUE THREAD

**Issue by happysalada** [fzakaria/mvn2nix#42]
Hi, thank you for the awesome package!

I'm trying to use this with a flake. Here is the [repo](github.com/ngi-nix/ipfs-search)

usage looks like
```
      nixpkgsFor = forAllSystems (system: import nixpkgs { inherit system; overlays = [ mvn2nix.overlay self.overlay ]; });
    in
    {
      # A Nixpkgs overlay.
      overlay = final: prev:
        with final;
        let
          mavenRepository = buildMavenRepositoryFromLockFile { file = ./mvn2nix-lock.json; };
```
on local it looks like it's working. But when I try to import that flake on my system (itself built with a flake) I get the following error
```
remote: error: undefined variable 'buildMavenRepositoryFromLockFile'
remote:
remote:        at /nix/store/cb2n4ii6rbvyx6pmr5fbbcxcqj5115ym-source/flake.nix:57:29:
remote:
remote:            57|           mavenRepository = buildMavenRepositoryFromLockFile { file = ./mvn2nix-lock.json; };
remote:              |                             ^
remote:            58|         in
remote: (use '--show-trace' to show detailed location information)
remote:       nixos-rebuild build --flake . -v
remote:       ^~~~~~~~~~~~~
```

I'm sure I am doing something wrong, but I can't really see what. I'm wondering if there isn't an additional step that is needed, where I need to import something.

**Comment by fzakaria** [comment]
I am not familiar with Flake --- I will ping the author of the flakes contribution @jonringer .

**Comment by happysalada** [comment]
hey sorry, I just discovered it was a problem in a completely different place, sorry for the trouble.
The flakes instructions are good.

for those people encountering a similar problem is that I used an overlay like it was recommended to build the packages. If you want to use the packages in a nixosModule, you will need that overlay as well. 

Sorry again for the trouble!
Thank you for the great software!
